lib/chef/resource/freebsd_package.rb

```diff
@@ -49,13 +49,11 @@ def ships_with_pkgng?
       end
 
       def assign_provider
-        @provider = if source.to_s =~ /^ports$/i
-                      Chef::Provider::Package::Freebsd::Port
-                    elsif supports_pkgng?
-                      Chef::Provider::Package::Freebsd::Pkgng
-                    else
-                      Chef::Provider::Package::Freebsd::Pkg
-                    end
+        @provider ||= if supports_pkgng?
+                        Chef::Provider::Package::Freebsd::Pkgng
+                      else
+                        Chef::Provider::Package::Freebsd::Pkg
+                      end
```

Comment:
this is still wrong, you should be using provides with a block on the pkgng and pkg providers instead.  also where did port go?